pkg/auth/rhsso_authz_handler.go

```diff
@@ -210,7 +210,8 @@ func (a *AuthzHandler) checkInfraEnvBasedAccess(id string, action Action, payloa
 	err = a.db.Select("cluster_id").First(&infraEnv, "id = ?", id).Error
 	if err != nil {
 		a.log.WithError(err).Errorf("failed to retrieve infra-env record %s", id)
-		return false, err
```

Comment:
For more correct behavior I would check if the error is gorm.RecordNotFound and return this. Otherwise, return the error (and 500)

Reply:
changed